app/Repositories/Eloquent/AddressRepository.php

```diff
@@ -0,0 +1,120 @@
+<?php 
+
+namespace App\Repositories\Eloquent;
+
+use App\User;
+use App\Models\Address;
+use Illuminate\Database\Eloquent\Model;  
+use Auth;
+
+
+class AddressRepository 
+{
+	/**      
+     * @var Model      
+     */     
+     protected $user;       
+
+    /**      
+     * ProductRepository constructor.      
+     *      
+     * @param Product $model      
+     */     
+    public function __construct(User $user)     
```

Comment:
Use UserRepository instead of the User model

Reply:
Sure, will do it.